### ISSUE TITLE
[Bug]: source AWR_DE defective

### ISSUE THREAD

**Issue by pehebo** [mampfes/hacs_waste_collection_schedule#2643]
### I Have A Problem With:

A specific source

### What's Your Problem

Since latest update of HACS source awr_de isnt loaded with error:
^ TypeError: can't compare datetime.datetime to datetime.date

Before everything workes fine.

### Source (if relevant)

awr_de

### Checklist Source Error

- [X] Use the example parameters for your source (often available in the documentation) (don't forget to restart Home Assistant after changing the configuration)
- [X] Checked that the website of your service provider is still working
- [ ] Tested my attributes on the service provider website (if possible)
- [X] I have tested with the latest version of the integration (master) (for HACS in the 3 dot menu of the integration click on "Redownload" and choose master as version)

### Checklist Sensor Error

- [X] Checked in the Home Assistant Calendar tab if the event names match the types names (if types argument is used)

### Required

- [X] I have searched past (closed AND opened) issues to see if this bug has already been reported, and it hasn't been.
- [X] I understand that people give their precious time for free, and thus I've done my very best to make this problem as easy as possible to investigate.

**Comment by 9H3SC** [comment]
Same here for source: awsh_de

**Comment by TheDuffman85** [comment]
Same issue here with source Abfall.IO / AbfallPlus

Logs:
2024-09-09 10:02:12.716 ERROR (SyncWorker_2) [custom_components.waste_collection_schedule.waste_collection_schedule.source_shell] fetch failed for source Abfall.IO / AbfallPlus:
Traceback (most recent call last):
  File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source_shell.py", line 145, in fetch
    entries = self._source.fetch()
              ^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source/abfall_io.py", line 177, in fetch
    dates = self._ics.convert(ics_file)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/service/ICS.py", line 48, in convert
    events: List[Any] = icalevents.events(
                        ^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/icalevents/icalevents.py", line 57, in events
    found_events += parse_events(
                    ^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/icalevents/icalparser.py", line 394, in parse_events
    elif e.end >= f and e.start <= t and is_not_exception(e.start):
         ^^^^^^^^^^
TypeError: can't compare datetime.datetime to datetime.date

**Comment by RobGri001** [comment]
Same issue here with source "Abfallwirtschaftsbetrieb des Landkreises Pfaffenhofen a.d.Ilm (AWP)"

**Comment by Jushav** [comment]
Same with Abfallwirtschaft Pforzheim

**Comment by cook2000** [comment]
Same here with generic ICS and infeo_at

`Logger: custom_components.waste_collection_schedule.waste_collection_schedule.source_shell
Quelle: custom_components/waste_collection_schedule/waste_collection_schedule/source_shell.py:147
Integration: Waste Collection Schedule ([Dokumentation](https://github.com/mampfes/hacs_waste_collection_schedule#readme))
Erstmals aufgetreten: 12:57:34 (1 Vorkommnisse)
Zuletzt protokolliert: 12:57:34

fetch failed for source infeo: Traceback (most recent call last): File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source_shell.py", line 145, in fetch entries = self._source.fetch() ^^^^^^^^^^^^^^^^^^^^ File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source/infeo_at.py", line 129, in fetch dates = self._ics.convert(response.text) ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/service/ICS.py", line 48, in convert events: List[Any] = icalevents.events( ^^^^^^^^^^^^^^^^^^ File "/usr/local/lib/python3.12/site-packages/icalevents/icalevents.py", line 57, in events found_events += parse_events( ^^^^^^^^^^^^^ File "/usr/local/lib/python3.12/site-packages/icalevents/icalparser.py", line 394, in parse_events elif e.end >= f and e.start <= t and is_not_exception(e.start): ^^^^^^^^^^ TypeError: can't compare datetime.datetime to datetime.date`

**Comment by M0nk3yOo** [comment]
Same for "Regioentsorgung Städteregion Aachen".
So it seems to be a general error.

**Comment by dm82m** [comment]
I am pretty sure that is has something todo with a python module within the individual instance. Maybe directly with the DateTime module. Why am I assume this? @cook2000 has the same HA version, same waste_collection_schedule version, same configuration as I. It works on my instance but not on his instance. We compared DateTime module and on my instance it is 4.9 and on his instance it is 5.5.

**Comment by M0nk3yOo** [comment]
Sounds good, because even if I downgrade the version, the issue persists

**Comment by dm82m** [comment]
I manually upgraded from DateTime 4.9 to DateTime 5.5 (current), also works, upgraded icalendar from 4.0.9 to 5.0.13 (current) and icalevents from 0.1.27 to 0.1.29 (current) - and it also works...

not sure if my above assumption is still valid... :D

**Comment by dm82m** [comment]
Ha I was right, more or less!

@cook2000 has icalevents 0.1.28, I downgraded to it and voila the error is there (see below). upgrade to icalevents 0.1.29 and it is back working.

so all people having this issue can do:
```
sudo docker exec -it homeassistant /bin/bash
pip upgrade DateTime icalendar icalevents
```

then restart home assistant core.

```
Dieser Fehler wurde von einer benutzerdefinierten Integration verursacht

Logger: custom_components.waste_collection_schedule.waste_collection_schedule.source_shell
Quelle: custom_components/waste_collection_schedule/waste_collection_schedule/source_shell.py:147
Integration: Waste Collection Schedule (Dokumentation)
Erstmals aufgetreten: 17:18:09 (1 Vorkommnisse)
Zuletzt protokolliert: 17:18:09

fetch failed for source infeo: Traceback (most recent call last): File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source_shell.py", line 145, in fetch entries = self._source.fetch() ^^^^^^^^^^^^^^^^^^^^ File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source/infeo_at.py", line 242, in fetch dates = self._ics.convert(response.text) ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/service/ICS.py", line 48, in convert events: List[Any] = icalevents.events( ^^^^^^^^^^^^^^^^^^ File "/usr/local/lib/python3.12/site-packages/icalevents/icalevents.py", line 57, in events found_events += parse_events( ^^^^^^^^^^^^^ File "/usr/local/lib/python3.12/site-packages/icalevents/icalparser.py", line 394, in parse_events elif e.end >= f and e.start <= t and is_not_exception(e.start): ^^^^^^^^^^ TypeError: can't compare datetime.datetime to datetime.date
```

**Comment by cook2000** [comment]
Thanks @dm82m !

Upgrading icalevents to 0.1.29 worked for me.

**Comment by pehebo** [comment]
And what about people using Home Assistant OS?Am 09.09.2024 um 17:24 schrieb Dirk ***@***.***>:﻿
Ha I was right, more or less!
@cook2000 has icalevents 0.1.28, I downgraded to it and voila the error is there (see below). upgrade to icalevents 0.1.29 and it is back working.
so all people having this issue can do:
sudo docker exec -it homeassistant /bin/bash
pip upgrade DateTime icalendar icalevents

then restart home assistant core.
Dieser Fehler wurde von einer benutzerdefinierten Integration verursacht

Logger: custom_components.waste_collection_schedule.waste_collection_schedule.source_shell
Quelle: custom_components/waste_collection_schedule/waste_collection_schedule/source_shell.py:147
Integration: Waste Collection Schedule (Dokumentation)
Erstmals aufgetreten: 17:18:09 (1 Vorkommnisse)
Zuletzt protokolliert: 17:18:09

fetch failed for source infeo: Traceback (most recent call last): File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source_shell.py", line 145, in fetch entries = self._source.fetch() ^^^^^^^^^^^^^^^^^^^^ File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source/infeo_at.py", line 242, in fetch dates = self._ics.convert(response.text) ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/service/ICS.py", line 48, in convert events: List[Any] = icalevents.events( ^^^^^^^^^^^^^^^^^^ File "/usr/local/lib/python3.12/site-packages/icalevents/icalevents.py", line 57, in events found_events += parse_events( ^^^^^^^^^^^^^ File "/usr/local/lib/python3.12/site-packages/icalevents/icalparser.py", line 394, in parse_events elif e.end >= f and e.start <= t and is_not_exception(e.start): ^^^^^^^^^^ TypeError: can't compare datetime.datetime to datetime.date

**Comment by dm82m** [comment]
I use home assistant os, so no idea what your question is about.

**Comment by pehebo** [comment]
I wonder how to upgrade to icalevents 0.1.29… 😀Am 09.09.2024 um 18:54 schrieb Dirk ***@***.***>:﻿
I use home assistant os, so no idea what your question is about.

**Comment by dm82m** [comment]
You need shell access, can be realized by ssh‘ing onto haos or use a addon. If you Are not familiar with it, just wait till the pr is integrated

**Comment by pehebo** [comment]
Oh, good idea! Didnt realize that…:)Am 09.09.2024 um 19:00 schrieb Dirk ***@***.***>:﻿
You need shell access, can be realized by ssh‘ing onto haos or use a addon. If you Are not familiar with it, just wait till the pr is integrated

**Comment by 5ila5** [comment]
This should hopefully be fixed when updating to the freshly released WCS version 2.2.0, Which sets the reuirements to be icalevents!=0.1.28
